llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -2344,6 +2344,21 @@ static unsigned getSHXADDShiftAmount(unsigned Opc) {
   }
 }
 
+// Returns the shift amount from a SHXADD.UW instruction. Returns 0 if the
+// instruction is not a SHXADD.UW.
+static unsigned getSHXADDUWShiftAmount(unsigned Opc) {
```

Comment:
I'd probably just merge this with the previous routine.  Reading the using code, I didn't even spot the difference in the name because I'd assumed it to be one routine.

Reply:
I almost made the same comment, but the original caller of the other function uses the return value to know if the instruction is a SHXADD. That caller would be incorrect for SHXADD_UW.